src/Serializer/RequestsSerializer.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace OpenEuropa\EPoetry\Serializer;
+
+use OpenEuropa\EPoetry\PropertyInfo\Extractor\MappingReflectionExtractor;
+use Symfony\Component\Serializer\Encoder\YamlEncoder;
+use Symfony\Component\Serializer\Normalizer\ArrayDenormalizer;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Class RequestsSerializer.
+ */
+final class RequestsSerializer implements SerializerInterface
```

Comment:
I can't think of any reason why this class should be final, I guess services can be theoretically extended.

Reply:
I think that most of the classes we produce should be using the keyword 'final'.
This class `RequestSerializer` shouldn't be extended. The flexibility should come from good abstractions, not from inheritance. 

This library provides 1 way to create requests, **that** shouldn't be altered.
I doubt that one would change that anyway. However, if someone wants to alter it, then they can do it and use proper dependency injection mechanism.

This '`final`' word forces users to inject the service in their own class properly.

Example, instead of doing:

```php
<?php

namespace Foo\Bar;

class MyCustomRequestSerializer extends RequestSerializer {
  // Stuff.
}
```

They should do:

```php
<?php

namespace Foo\Bar;

class MyCustomRequestSerializer implements SerializerInterface {
  public function __construct(RequestsSerializer $serializer) {
    $this->serializer = $serializer;
  }
}
```

Reply:
Ok I see your point, we can keep it.

Reply:
Cool :-)